include/yaramod/types/hex_string.h

```diff
@@ -116,35 +121,37 @@ class HexStringNibble : public HexStringUnit
 public:
 	/// @name Constructors
 	/// @{
-	explicit HexStringNibble(std::uint8_t value) : HexStringUnit(Type::Nibble), _value(value) {}
+	explicit HexStringNibble(TokenIt value) : HexStringUnit(Type::Nibble), _value(value) {}
 	/// @}
 
 	/// @name Virtual methods
 	/// @{
 	virtual std::string getText() const override
 	{
-		assert(_value <= 0xf);
+		auto value = getValue();
+		assert(value <= 0xf);
 
-		if (_value <= 9)
-			return std::string(1, _value + '0');
+		if (value <= 9)
+			return std::string(1, value + '0');
 		else
-			return std::string(1, _value - 10 + 'A');
+			return std::string(1, value - 10 + 'A');
 	}
 
 	/// @name Getters
 	/// @{
 	std::uint8_t getValue() const
 	{
-		assert(_value <= 0xf);
-		return _value;
+		auto output = _value->getInt();
+		assert(output <= 0xf);
+		return output;
 	}
 	/// @}
 
 	virtual std::size_t getLength() const override { return 1; }
 	/// @}
 
 private:
-	std::uint8_t _value; ///< Value of the nibble
+	TokenIt _value; ///< Value of the nibble, int
```

Comment:
It was supposed to indicate what type is stored in the variant inside `Token`. Here it is wrong, because the type is std::string in this case. Thank you for noticing.